concent_api/core/validation.py

```diff
@@ -373,3 +369,36 @@ def validate_file_transfer_token(file_transfer_token: message.concents.FileTrans
     categories = [file_info['category'] for file_info in file_transfer_token.files]
     if len(set(categories)) != len(categories):
         raise FileTransferTokenError("'category' field must be unique across FileInfo list.", ErrorCode.MESSAGE_FILES_CATEGORY_NOT_UNIQUE)
+
+
+def validate_secure_hash_algorithm(checksum: str):
+    if not isinstance(checksum, str):
+        raise HashingAlgorithmError(
+            "'checksum' must be a string.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_TYPE
+        )
+
+    if len(checksum) == 0 or checksum.isspace():
```

Comment:
why blending those two together? Wouldn't it be better if we had two separete check with two separate errors?

Reply:
Not sure it's pretty simple check :P